### ISSUE TITLE
Table.write(format='ascii.ecsv') ignores formats keyword and column attribute

### ISSUE THREAD

**Issue by abostroem** [astropy/astropy#15000]
### Description

When writing a table, the format of the output can be specified with either the `formats` keyword or the `Column.format` attribute. However, when the output file format is `ascii.ecsv` both of these are ignored.

### Expected behavior

That the output file would be formatted as specified (see How to Reproduce for example)

### How to Reproduce

```python
from astropy.table import Table
import numpy as np
x = np.arange(10)/13
tbdata = Table([x], names=['x'])
formats={'x':'%1.2f'}
tbdata.write('test.ecsv', formats=formats)
```

Produces:
```
# %ECSV 1.0
# ---
# datatype:
# - {name: x, datatype: float64}
# schema: astropy-2.0
x
0.0
0.07692307692307693
...
0.6923076923076923
```

while 
```python
tbdata.write('test.asc', format='ascii', formats=formats)```

Produces:
```
x
0.00
0.08
...
0.69
```

this same behavior is observed if instead of providing the `formats` keyword the column format attribute is set
```python
tbdata['x'].format='%1.2f'
```

### Versions

macOS-10.16-x86_64-i386-64bit
Python 3.9.13 (main, Aug 25 2022, 18:29:29) 
[Clang 12.0.0 ]
astropy 5.3
Numpy 1.21.5
pyerfa 2.0.0
Scipy 1.9.1
Matplotlib 3.5.2

**Comment by pllim** [comment]
Is this a duplicate of https://github.com/astropy/astropy/issues/14461 ?

**Comment by abostroem** [comment]
yes, thanks for pointing that out. Could this "attribute" of ECSV be documented somewhere (apologies if I missed it). I'll note that ECSV is the only ascii format that outputs metadata, which is why I use it, but I can hack together an ECSV and CSV file.

**Comment by eerovaher** [comment]
The very first sentence in the ECSV documentation emphasizes that it is possible to convert a `Table` to an ECSV file and back without loss of information:
https://github.com/astropy/astropy/blob/1d3b81abb8db3c935924b603daff2c47a6f4a9df/docs/io/ascii/ecsv.rst?plain=1#L6-L10
This would not be possible if ECSV would not ignore any formatting metadata that might be present in the table. But it might be good if the ECSV writer would raise a warning or an error if it is called with the `formats` keyword.
